src/dex/maverick-v2/maverick-v2.ts

```diff
@@ -193,6 +185,9 @@ export class MaverickV2 extends SimpleExchange implements IDex<MaverickV2Data> {
             return MAV_V2_BASE_GAS_COST + MAV_V2_TICK_GAS_COST * Number(t);
           });
 
+          // If pool isn't fully initialized, returns null
+          if (unit === 1n) return null;
```

Comment:
What if unit is `0` (when error happened)?

Reply:
Also, wasn't able to find where we return `1` if pool is not initialized. Could you clarify where it's done?

Reply:
This fix wont help. There is a problem with pricing calculation and real amount of reserves - MaverickV2 uses ticks for calculating prices, but reserves in ticks are virtual and does not represent real reserves in pool. 

That could leads  to incorrect pricing on BUY side (ex. user swaps 1 USDT and receives 0.00001 USDC, but pricing in this pool will be ~1:1)

Reply:
removed this fix